op-batcher/batcher/driver.go

```diff
@@ -159,11 +158,11 @@ func (l *BatchSubmitter) Start() error {
 	return nil
 }
 
-func (l *BatchSubmitter) StopIfRunning() {
-	_ = l.Stop()
+func (l *BatchSubmitter) StopIfRunning(ctx context.Context) {
+	_ = l.Stop(ctx)
 }
 
-func (l *BatchSubmitter) Stop() error {
+func (l *BatchSubmitter) Stop(ctx context.Context) error {
```

Comment:
The `ctx` parameter in is no longer used anywhere. The original purpose of passing this in was so that the passed in context could force cancel this call if needed, and had a small go routine that would cancel the "kill" context if the parameter was canceled: https://github.com/BrianBland/optimism/pull/1/commits/6cce718f7afdf5423bda6f2cc515c6d08d75d505#diff-c734d1296b2fd691221b92df3edf09c7533c507a74c2316117745c75c3ad5776R168-R175

did we decide that was a bad idea?

Reply:
If we want to add an (optional) timeout parameter to the stop rpc, it could still make sense to have this context. Then the rpc could use the provided timeout parameter to create a `context.WithTimeout`.

Reply:
Let me add that back - I might have accidentally lost that to rebases/merge conflicts 😅